modelstore/storage/blob_storage.py

```diff
@@ -145,14 +132,49 @@ def download(self, local_path: str, domain: str, model_id: str = None):
         if model_id is None:
             model_domain = get_domain_path(self.root_prefix, domain)
             model_meta = self._read_json_object(model_domain)
-            logger.info("Latest model is: %s", model_meta["model"]["model_id"])
-        else:
-            model_meta_path = self._get_metadata_path(domain, model_id)
-            # Note: this will fail if the model does not exist (needs a more informative exception)
-            model_meta = self._read_json_object(model_meta_path)
+            model_id = model_meta["model"]["model_id"]
+            logger.info("Latest model is: %s", model_id)
+
+        model_meta = self.get_meta_data(domain, model_id)
         storage_path = self._get_storage_location(model_meta["storage"])
         return self._pull(storage_path, local_path)
 
+    def delete_model(
+        self, domain: str, model_id: str, meta_data: dict, skip_prompt: bool = False
+    ):
+        """Deletes a model artifact from storage. Other than the artifact itself
+        being deleted:
+        - The model is unset from all states.
+        - The model will no longer be returned when using list_models()
+        - One meta data file is preserved, using the reserved DELETED state"""
+        if not skip_prompt:
+            message = f"Delete model from domain={domain} with model_id={model_id}?"
+            if not click.confirm(message):
+                logger.info("Aborting; not deleting model")
+                return
+
+        # Delete the artifact itself
+        storage_path = self._get_storage_location(meta_data["storage"])
+        self._remove(storage_path)
+
+        # Set the model as deleted in the meta data by unsetting it from
+        # all custom states, setting it to a reserved state, and then deleting
+        # the main meta-data file
+        logger.debug("Removing model from all states %s=%s", domain, model_id)
+        for state_name in self.list_model_states():
+            self.unset_model_state(domain, model_id, state_name)
+
+        logger.debug("Setting to state=deleted %s=%s", domain, model_id)
+        self.set_model_state(domain, model_id, ReservedModelStates.DELETED.value)
+
+        logger.debug("Deleting meta-data for %s=%s", domain, model_id)
+        remote_path = self._get_metadata_path(domain, model_id)
+        self._remove(remote_path)
+
+        # @TODO (future): the model that is being deleted may be also set
+        # as the "latest" model in a domain; this will cause download() to fail
+        # if a model_id is not provided
```

Comment:
I'm leaving this out for now - I want to, more generally, re-think how the "latest" model is stored. I'd like to use a `ReservedModelState` for it instead of hiding it inside of the domain information.